### PR TITLE
PR-134 Updates

### PR DESCRIPTION
**Pull request by 5arias**
Given that @mladen107 hasn't had the time to make the updates for his PR, I went ahead and made them since I need support for adding initial files in the gallery as well.

I removed the onClick functionality in src/filename.jsx for the time being. This is a feature that I'll be adding in for my project in the somewhat near future, however I believe it should include status checks, onClick and href support - features which I didn't have time to write last night.

I also rearranged the order of the isCancelable status array to reflect the status workflow detailed in the [stats & status documentation](https://docs.fineuploader.com/branch/master/features/statistics-and-status-updates.html)

To answer @rnicholus comment regarding the statusEnum.SUBMITTING status being included, I agree with @mladen107 that this status should be included because it describes a cancelable event (validation checks occur during this event).

### CODE REVIEW

**Comment by mladen107** [comment]
Sorry for not updating my PR. At the end, I needed much more complex UI so I wrote my own from scratch. I agree that onClick should be separate pull request.

File: src/gallery/index.jsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ class Gallery extends Component {
         super(props)
 
         this.state = {
-            visibleFiles: []
+            visibleFiles: props.uploader.methods.getUploads()
```
Comment (rnicholus) [inline src/gallery/index.jsx]: Let's limit each file entry to the `id` and optionally `fromServer`.

**Review by rnicholus — CHANGES_REQUESTED** [review]
One change, and this really needs more tests, especially on the impacted components.

File: src/gallery/index.jsx (hunk 1 above)
Comment (5arias) [reply src/gallery/index.jsx]: Sounds good and I agree about the tests. I probably won't be able to get to it until next week. Thanks!

**Comment by 5arias** [comment]
@mladen107 All good, I appreciate your initial work on it!

**Comment by 5arias** [comment]
@rnicholus I refactored to limit the initial file objects to ID and Status. I haven't had the time to work on making `fromServer` optional. 

Also what other tests would you like to see?

**Comment by rnicholus** [comment]
Can someone remind me what problem we are trying to solve here? Seems like a lot of changes as well...

**Comment by rnicholus** [comment]
closing due to inactivity

**Pull request closed without merging**